x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportGetCheckpointNodeAction.java

```diff
@@ -83,17 +86,27 @@ protected static void getGlobalCheckpoints(
                     return;
                 }
             }
-            final IndexService indexService = indicesService.indexServiceSafe(shardId.getIndex());
-            final IndexShard indexShard = indexService.getShard(shardId.id());
 
-            checkpointsByIndexOfThisNode.computeIfAbsent(shardId.getIndexName(), k -> {
-                long[] seqNumbers = new long[indexService.getIndexSettings().getNumberOfShards()];
-                Arrays.fill(seqNumbers, SequenceNumbers.UNASSIGNED_SEQ_NO);
-                return seqNumbers;
-            });
-            checkpointsByIndexOfThisNode.get(shardId.getIndexName())[shardId.getId()] = indexShard.seqNoStats().getGlobalCheckpoint();
-            ++numProcessedShards;
+            try {
+                final IndexService indexService = indicesService.indexServiceSafe(shardId.getIndex());
+                final IndexShard indexShard = indexService.getShard(shardId.id());
+
+                checkpointsByIndexOfThisNode.computeIfAbsent(shardId.getIndexName(), k -> {
+                    long[] seqNumbers = new long[indexService.getIndexSettings().getNumberOfShards()];
+                    Arrays.fill(seqNumbers, SequenceNumbers.UNASSIGNED_SEQ_NO);
+                    return seqNumbers;
+                });
+                checkpointsByIndexOfThisNode.get(shardId.getIndexName())[shardId.getId()] = indexShard.seqNoStats().getGlobalCheckpoint();
+                ++numProcessedShards;
+            } catch (Exception e) {
```

Comment:
The PR description says that the `IndexNotFoundException` is forwarded but in fact **every** exception is forwarded here.
I think the code is ok, just please update PR description.